vnext/JSI/Shared/ChakraRuntimeArgs.h

```diff
@@ -2,49 +2,71 @@
 // Licensed under the MIT License.
 
 #pragma once
-#include <jsi/jsi.h>
-#include "ScriptStore.h"
+
+#include "PreparedJavaScriptStore.h"
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <utility>
 
 namespace facebook::react {
 class MemoryTracker;
-class MessageQueueThread;
-} // namespace facebook::react
+}
 
 namespace Microsoft::JSI {
 
-enum class LogLevel {
-  Trace = 0,
-  Info = 1,
-  Warning = 2,
-  Error = 3,
-  Fatal = 4,
-};
-
-using Logger = std::function<void(const char *message, LogLevel logLevel)>;
-
 struct ChakraRuntimeArgs {
-  bool enableJITCompilation{true};
+ public:
+  /**
+   * @enum ExecutablePageAllocationPolicy Indicates whether a ChakraRuntime
+   * is allowed to allocate executatble pages.
+   */
+  enum class ExecutablePageAllocationPolicy {
+    /**
+     * @var ChakraRuntime will not allocate any executable code pages. This
+     * also implies that Native Code generation will be turned off. Note that
+     * this will break JavaScript stack decoding in tools like WPA, since
+     * they rely on allocation of unique thunks to interpret each function and
+     * allocation of those thunks will be disabled as well
+     */
+    DisableAll,
+    /**
+     * @var ChakraRuntime will not generate native code.
+     */
+    DisableNativeCodeGeneration,
+    /**
+     * @var ChakraRuntime is free to generate executable pages.
+     */
+    EnableAll
+  };
 
-  Logger loggingCallback{};
+  /**
+   * @brief ChakraRuntime's policy for allocating executable pages.
+   */
+  ExecutablePageAllocationPolicy m_executablePageAllocationPolicy{ExecutablePageAllocationPolicy::EnableAll};
```

Comment:
>m_ [](start = 33, length = 2)

nit: I usually don't see the "m_" prefix for structs, only for private class members

Reply:
I don't have strong preferences here so if people do not like the "m_" prefix I can remove them.

---
In reply to: [345458052](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345458052) [](ancestors = 345458052)